src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs

```diff
@@ -2206,12 +2206,51 @@ internal int StartSSHProcess(
             var context = Runspaces.LocalPipeline.GetExecutionContextFromTLS();
             if (context != null)
             {
-                var cmdInfo = context.CommandDiscovery.LookupCommandInfo(sshCommand, CommandOrigin.Internal) as ApplicationInfo;
+                var cmdInfo = CommandDiscovery.LookupCommandInfo(
+                    sshCommand,
+                    CommandTypes.Application,
+                    SearchResolutionOptions.None,
+                    CommandOrigin.Internal,
+                    context) as ApplicationInfo;
                 if (cmdInfo != null)
                 {
                     filePath = cmdInfo.Path;
                 }
             }
+            else
+            {
+                // A Runspace may not be present in the TLS in SDK hosted apps
+                // or if running in another thread without a Runspace. While
+                // ProcessStartInfo can lookup the full path in PATH, we set
+                // the cwd to the parent directory of the found executable so
+                // we do a manual lookup here.
```

Comment:
I don't see where we _explicitly_ set the cwd. So we could exclude manual lookup and do here simply:
```c#
    filePath = sshCommand;
```

Reply:
Thanks for the clarification! Now it's clear that by `cwd` you mean `WorkingDirectory` property.
If we explicitly set WorkingDirectory in line 2325, then according to the [documentation](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.workingdirectory?view=net-9.0), this affects the app search only if UseShellExecute is true but we set it to _false_ in line 2327.

Reply:
The current logic `CommandDiscovery.LookupCommandInfo(...)` doesn't look at the current working directory. It only returns an application command that can be found in PATH. However, `ProcessStartInfo` first looks at the current working directory (the USERPROFILE path by default?) and then search in PATH. I guess that's why we try to resolve the `ssh` command before starting the process.

@jborean93 Can you please update the "ProcessStartInfo" part of the comment?

Reply:
@daxian-dbw Is that a problem?

LookupCommandInfo can also create a problem. For example, if you create an ssh file in your home folder on Linux without the x attribute.

In the context of this PR, the behavior of the system call (ProcessStartInfo) seems more correct. In reality, the search will be performed in the following order:
- the directory where pwsh is located.
- the default system directory.
- paths from PATHs.

Reply:
In my testing, `ProcessStartInfo` looks at the process-level working directory when resolving a command. For example, I create the file `abc.exe` under `e:\temp` and then start `pwsh` from CMD in that folder to make it the process-level working directory. Then running the following code in the `pwsh` session will result in the error `"... The specified executable is not a valid application for this OS platform."`. It indicates 'ProcessStartInfo' searched the process-level working directory for the specified command name. It actually searches that first before PATH.

```powershell
$startInfo = New-Object System.Diagnostics.ProcessStartInfo
$startInfo.FileName = 'abc.exe'
$process = New-Object System.Diagnostics.Process
$process.StartInfo = $startInfo
$process.Start()
```

`LookupCommandInfo` doesn't search the process-level working directory, and it's safer to keep that behavior.

Reply:
How about changing to the following?

```suggestion
                // A Runspace may not be present in the TLS in SDK hosted apps
                // or if running in another thread without a Runspace. While
                // 'ProcessStartInfo' can lookup the full path in PATH, it searches
                // the process' working directory first. 'LookupCommandInfo' does
                // not search the process' working directory and we want to keep that
                // behavior. So, we do a manual lookup here only in PATH.
```

Reply:
That's true. Can you please update the comment to capture both of those points? Thank you!